__tests__/app.test.js

```diff
@@ -0,0 +1,193 @@
+            slug: "cats",
+          },
+          {
+            description: "what books are made of",
+            slug: "paper",
+          },
+        ]);
+      });
+  });
+  test("should respond with 404 path not found", () => {
+    return request(app)
+      .get("/api/topicz")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("404 - path not found");
+      });
+  });
+});
+
+describe("getArticles", () => {
+  test("Responds with the required first article", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((articles) => {
+        expect(articles.body).toEqual(
+          expect.objectContaining({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: 1,
+            body: expect.any(String),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+          })
+        );
+      });
+  });
+  test("should respond with 404 path not found", () => {
+    return request(app)
+      .get("/api/articles/112222222222222222")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("404 - article not found");
+      });
+  });
+  test("should respond with 400 ID not a number", () => {
+    return request(app)
+      .get("/api/articles/1n2n3n4n")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("400 - article ID must be a number!");
+      });
+  });
+});
+
+describe("getUsers", () => {
+  test("should return an array of usernames", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((response) => {
+        expect(response.body).toBeInstanceOf(Array);
+        response.body.forEach((user) => {
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+
+  test("should respond with 404 path not found", () => {
+    return request(app)
+      .get("/api/userz")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("404 - path not found");
+      });
+  });
+});
+
+describe("getArticles", () => {
+  test("Responds with all articles in descending order", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        expect(response.body).toBeSortedBy("created_at", { descending: true });
+        response.body.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              body: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+            })
```

Comment:
again, include check for length of the articles array

Reply:
also, we don not need the body when returning the whole array

Reply:
rectified